pyquil/tests/test_parser.py

```diff
@@ -170,14 +171,47 @@ def test_pragma():
     _test('PRAGMA NO-NOISE', Pragma('NO-NOISE'))
 
 
+def test_include():
+    parse('INCLUDE "somefile"')
+
+
+def test_expression_functions():
+    parse('RX(cos(%x)) 1')
+    parse('RX(sin(%x)) 1')
+    parse('RX(exp(%x)) 1')
+    parse('RX(cis(%x)) 1')
+    parse('RX(sqrt(%x)) 1')
+
+
 def test_invalid():
     with pytest.raises(RuntimeError):
         parse("H X")
 
 
+def test_dynamic_parameter():
+    with pytest.raises(NotImplementedError):
+        parse("RX([1]) 2")
+
+    with pytest.raises(NotImplementedError):
+        parse("RX([1-10]) 2")
+
+
 def test_empty_program():
     _test("")
 
 
+def test_parse_variables():
```

Comment:
Also add some tests for different types of variables (not starting with number, ending with underscore, etc etc)